### PR TITLE
Added a _getting started_ guide to set up a blank environment that can be used to deploy services.

### PR DESCRIPTION
**Pull request by soenkeliebau**

### CODE REVIEW

File: README.adoc [hunk 1]
```diff
@@ -1,7 +1,288 @@
...
+
+* *Orchestrator*: A single server which is used to run components that are needed only once
+* *Worker*: One or more servers that are intended to run the actual workloads (HDFS, Kafka, ...) later on
+
+NOTE: Going forward, we will assume that you are installing this on several machines, but most commands should work the same in a single-machine scenario.
+We will alert you if something needs to be done differently when running everything on the same machine.
+
+
+.By the end of this guide, the environment that you have set up will look like this
+image::images/quickstart_architecture.png[]
```
Comment (ohessel) [inline README.adoc:28]: Image is not part of the PR - please add.

Comment (backstreetkiwi) [reply README.adoc:28]: I'll review this by using the guide to work on Issue https://github.com/stackabletech/t2/issues/11

File: README.adoc [hunk 2]
```diff
@@ -1,7 +1,288 @@
...
+  names:
+    plural: repositories
+    singular: repository
+    kind: Repository
+    shortNames:
+    - repo
+EOF
+----
+
+You can either host your own repository, if you so choose, or specify our public repository for convenience.
```
Comment (maltesander) [inline README.adoc:177]: You can either host your own repository or specify our public repository for convenience.

File: README.adoc [hunk 3]
```diff
@@ -1,7 +1,288 @@
...
+  name: stackablepublic
+spec:
+  repo_type: StackableRepo
+  properties:
+    url: https://repo.stackable.tech/repository/packages/
+EOF
+----
+
+=== Install Operators
+Depending on which products you want to manage with the Stackable platform you will want to install at least one of our operators.
```
Comment (maltesander) [inline README.adoc:194]: Depending on the products you want to manage with the Stackable platform, you need to install our corresponding operators.

**Review by maltesander — COMMENTED** [review]
I just commented on some language. 
There are lots of "will" / "do not" abbreviations. While these may keep a certain closer relationship to the reader, they should not be used in formal language / documents. Although it feels more "young" or "start-up-like" so its maybe not even bad :-). I think we should agree on a level of formality (with or without these abbreviations) in documents like this.

File: README.adoc [hunk 4]
```diff
@@ -1,7 +1,288 @@
...
+|Apache Kafka
+|stackable-kafka-operator-server
+
+|Apache Spark
+|stackable-spark-operator-server
+
+|Apache Zookeeper
+|stackable-zookeeper-operator-server
+|===
+
```
Comment (ohessel) [inline README.adoc:210]: I'd add an example here - as done below for the agent or add a link to the respective project in github

===== RHEL

    *coming soon*

===== Ubuntu

    apt-get install stackable-spark-operator-server

Comment (fwaibel) [reply README.adoc:210]: 👍

File: README.adoc [hunk 5]
```diff
@@ -1,7 +1,288 @@
...
+Put the following content in `/usr/lib/systemd/system/stackable-agent.service.d/kubeconfig.conf`:
+
+----
+Environment="KUBECONFIG=/path/to/kubeconfig"
+----
+
+==== Starting the Agent
+The agent can be started like any regular systemd service by running
+
+` systemctl start stackable-agent`
```
Comment (fwaibel) [inline README.adoc:269]: The extra space before systemctl breaks the formatting

File: README.adoc [hunk 6]
```diff
@@ -1,7 +1,288 @@
...
+
+To enable it to be started at every boot:
+` systemctl enable stackable-agent`
```
Comment (fwaibel) [inline README.adoc:272]: The extra space before systemctl breaks the formatting

File: README.adoc [hunk 7]
```diff
@@ -1,7 +1,288 @@
...
+If your Kubernetes is not configured to auto-approve these CSRs, you will need to manually approve that request before the agent can start.
+
+You can do this by running `kubectl certificate approve <agent-hostname>-tls` on the orchestrator server after starting the agent.
+
+If your Kubernetes does not offer certificate signing services, you can alternatively also manually generate a keypair and sign a certificate with a proper authority.
+These files will then need to be configured in the agent config file.
+
+=== Health Check
+After performing the steps outlined above you should have an empty Stackable Platform up and running.
+When you run `kubectl nodes get` on the orchestrator, you should see the objects that all your agents added to indicate their readiness.
```
Comment (fwaibel) [inline README.adoc:285]: The arguments seem to be permuted: `kubectl get nodes` worked fine.

File: README.adoc [hunk 8]
```diff
@@ -1,7 +1,288 @@
...
+===== Centos/RHEL
+
+*coming soon*
+
+=== Install Kubernetes
+Usually the installation and configuration of Kubernetes is a full-blown project all of its own, so it might seem a bit weird to find this tucked away in a little section of the quickstart guide like this.
+However, the Stackable Platform only uses a very limited subset of the full Kubernetes features, so for our purposes we could actually get away with just running a https://kubernetes.io/docs/reference/command-line-tools-reference/kube-apiserver/[kube-apiserver] and leaving out the entire rest of the https://kubernetes.io/docs/concepts/overview/components/[control-plane].
+This makes the deployment a lot simpler.
+
+That being said, Stackable will work with any Kubernetes, please feel free to skip this chapter if you already have a working Kubernetes cluster, or want to use a managed Kubernetes offering like the ones provided by https://www.ionos.com/enterprise-cloud/managed-kubernetes[IONOS], https://cloud.google.com/kubernetes-engine[Google] or https://docs.microsoft.com/en-us/azure/aks/[Azure].
```
Comment (lfrancke) [inline README.adoc:59]: Are we sure they allow custom nodes?
(IONOS do but the others?)
Maybe add a note that we haven't tested them.

File: README.adoc [hunk 9]
```diff
@@ -1,7 +1,288 @@
...
+
+If you choose to follow one of the options mentioned above, feel free to skip ahead to https://xxx[xxx], as you won't need the information in the rest of this chapter.
```
Comment (lfrancke) [inline README.adoc:61]: ....

File: README.adoc [hunk 10]
```diff
@@ -1,7 +1,288 @@
...
+
+Since you are still reading this we'll assume that you don't have easy access to an existing Kubernetes and instead install one.
```
Comment (lfrancke) [inline README.adoc:63]: This reads weird "instead install one". Maybe we can just remove this sentence?

File: README.adoc [hunk 11]
```diff
@@ -1,7 +1,288 @@
...
+There are numerous possibilities to go about this, but we will focus on https://k3s.io/[K3S], a lightweight Kubernetes distribution that is provided by https://rancher.com/[Rancher].
+
+Run the following command on your _orchestrator_ server:
```
Comment (lfrancke) [inline README.adoc:66]: You used `node` above instead of `server`

File: README.adoc [hunk 12]
```diff
@@ -1,7 +1,288 @@
...
+
+    curl -sfL https://get.k3s.io | sh -
```
Comment (lfrancke) [inline README.adoc:68]: Maybe mention something about versions?
We only tested Kubernetes 1.18 and above.

File: README.adoc [hunk 13]
```diff
@@ -1,7 +1,288 @@
...
+
+----
+Kubernetes control plane is running at https://127.0.0.1:6443
+CoreDNS is running at https://127.0.0.1:6443/api/v1/namespaces/kube-system/services/kube-dns:dns/proxy
+Metrics-server is running at https://127.0.0.1:6443/api/v1/namespaces/kube-system/services/https:metrics-server:/proxy
+
+To further debug and diagnose cluster problems, use 'kubectl cluster-info dump'.
+----
+
+==== Distribute KubeConfig
```
Comment (lfrancke) [inline README.adoc:87]: You use KubeConfig and Kubeconfig. Not sure which is correct but it'd be best to be consistent.

File: README.adoc [hunk 14]
```diff
@@ -1,7 +1,288 @@
...
+To avoid having to configure these repositories on every agent they are instead specified in Kubernetes and retrieved from there by the agent.
+
+In order to allow creating a repository, you'll have to create the CRD for repositories in your freshly installed Kubernetes cluster.
+The CRD looks like this:
+
+----
+apiVersion: apiextensions.k8s.io/v1
+kind: CustomResourceDefinition
+metadata:
+  name: repositories.stable.stackable.de
```
Comment (lfrancke) [inline README.adoc:111]: We should change this to stackable.tech (also in the group below)
This'll probably need a follow-up issue in the agent?

File: README.adoc [hunk 15]
```diff
@@ -1,7 +1,288 @@
...
+----
+systemctl status stackable-zookeeper-operator-server
+----
+
+This should report the service as being loaded and enabled.
+
+=== Add Nodes to Kubernetes
+
+==== Install Agent
+The distributed component of the Stackable Platform is the agent, which needs to be installed on all managed servers and configured to have access to Kubernetes.
```
Comment (lfrancke) [inline README.adoc:239]: "The distributed component" reads weid and doesn't make sense to me.

```suggestion
Tthe agent needs to be installed on all servers that should run services and they need to be configured to have access to Kubernetes.
```
Or similar

File: README.adoc [hunk 16]
```diff
@@ -1,7 +1,288 @@
...
+
+Installing the agent can be done from our rpm and deb package repositories, which we enabled when preparing the nodes above.
```
Comment (lfrancke) [inline README.adoc:241]:
```suggestion
Installing the agent can be done from our RPM and deb package repositories, which we enabled when preparing the nodes above.
```

File: README.adoc [hunk 17]
```diff
@@ -1,7 +1,288 @@
...
+===== Ubuntu
+
+    apt-get install stackable-agent
+
+==== Configuring the Agent
+When installing from packages the agent config file is created in '/etc/stackable-agent/agent.conf'.
+
+The Agent takes a few configuration options, please refer to https://github.com/stackabletech/agent#command-line-parameters[the agent documentation] for more details.
+
+If you have provided a KUBECONFIG for the root user in `/root/.kube/config` then the agent will use this, if you decided against a _global_ KUBECONFIG, you will need to specify where the config should be read from.
```
Comment (lfrancke) [inline README.adoc:257]: Again consistency, KUBECONFIG vs KubeConfig vs Kubeconfig

File: README.adoc [hunk 18]
```diff
@@ -1,7 +1,288 @@
...
+
+Please see below for extra steps that need to be taken to enable this config.
+
+=== Specify a Repository
+Stackable downloads packages from repositories, which the agents need to know about.
+To avoid having to configure these repositories on every agent they are instead specified in Kubernetes and retrieved from there by the agent.
+
+In order to allow creating a repository, you'll have to create the CRD for repositories in your freshly installed Kubernetes cluster.
+The CRD looks like this:
+
```
Comment (lfrancke) [inline README.adoc:106]:
```suggestion

[source,yaml]
```

File: README.adoc [hunk 19]
```diff
@@ -1,7 +1,288 @@
...
+  names:
+    plural: repositories
+    singular: repository
+    kind: Repository
+    shortNames:
+    - repo
+----
+
+You can choose whatever way is most convenient for you to apply this CRD to your cluster, one possible way is using `kubectl` from the command line:
+
```
Comment (lfrancke) [inline README.adoc:141]:
```suggestion

[source,bash]
```

Not sure if that works

File: README.adoc [hunk 20]
```diff
@@ -1,7 +1,288 @@
...
+    singular: repository
+    kind: Repository
+    shortNames:
+    - repo
+EOF
+----
+
+You can either host your own repository, if you so choose, or specify our public repository for convenience.
+The specification for our repository is shown below and can be applied with `kubectl` just like the definition above:
+
```
Comment (lfrancke) [inline README.adoc:179]:
```suggestion

[source,bash]
```
see above

**Review by lfrancke — CHANGES_REQUESTED** [review]

File: README.adoc (hunk 14 above)
Comment (soenkeliebau) [reply README.adoc:111]: I'll need to check in the agent, but most probably, yes. Can we leave as is for now to get this merged?

File: README.adoc (hunk 17 above)
Comment (soenkeliebau) [reply README.adoc:257]: KUBECONFIG in this case refers to an environment variable, which by convention should be spelled in all caps I believe. I've fixed the other occurrences to all lower case which seems to be consistend with the kubernetes config.
https://kubernetes.io/docs/concepts/configuration/organize-cluster-access-kubeconfig/

File: README.adoc [hunk 21]
```diff
@@ -56,14 +57,33 @@ Usually the installation and configuration of Kubernetes is a full-blown project
 However, the Stackable Platform only uses a very limited subset of the full Kubernetes features, so for our purposes we could actually get away with just running a https://kubernetes.io/docs/reference/command-line-tools-reference/kube-apiserver/[kube-apiserver] and leaving out the entire rest of the https://kubernetes.io/docs/concepts/overview/components/[control-plane].
 This makes the deployment a lot simpler.
 
+NODE: We have tested our stack with Kubernetes 1.18 and above
```
Comment (lfrancke) [inline README.adoc:60]:
```suggestion
NOTE: We have tested our stack with Kubernetes 1.18 and above
```

File: README.adoc [hunk 22]
```diff
@@ -254,7 +284,7 @@ When installing from packages the agent config file is created in '/etc/stackabl
 
 The Agent takes a few configuration options, please refer to https://github.com/stackabletech/agent#command-line-parameters[the agent documentation] for more details.
 
-If you have provided a KUBECONFIG for the root user in `/root/.kube/config` then the agent will use this, if you decided against a _global_ KUBECONFIG, you will need to specify where the config should be read from.
+If you have provided a kubeconfig for the root user in `/root/.kube/config` then the agent will use this, if you decided against a _global_ `KUBECONFIG` enwironment variable, you will need to specify where the config should be read from.
```
Comment (lfrancke) [inline README.adoc:287]:
```suggestion
If you have provided a kubeconfig for the root user in `/root/.kube/config` then the agent will use this, if you decided against a _global_ `KUBECONFIG` environment variable, you will need to specify where the config should be read from.
```

File: README.adoc [hunk 23]
```diff
@@ -85,13 +105,13 @@ To further debug and diagnose cluster problems, use 'kubectl cluster-info dump'.
 ----
 
 ==== Distribute KubeConfig
```
Comment (lfrancke) [inline README.adoc:107]: This is still off?

**Review by lfrancke — CHANGES_REQUESTED** [review]

**Review by lfrancke — APPROVED** [review]

**Pull request merged**